src/messenger/messenger.cpp

```diff
@@ -211,16 +203,16 @@ void Messenger::Terminate() {
 }
 
 ConnectionId Messenger::MakeConnection(const ConnectionDestination &target, std::optional<std::string> identity) {
```

Comment:
Is it better to return a unique_ptr here? It makes easier for caller of this to own a `ConnectionId` I guess

Reply:
I am having this use case in mind:
My pilotmanager will want to create a connection with the messenger, but not necessary at the construction (or it will need to somehow reset a connection during the lifetime of the pilotmanager). 
This means the pilotmanager preferably needs to have a member field to a connectionid, but returning a ConnectionId here kind of enforces the lifetime to be the lifetime of the enclosing function call. 

Or is there a better way for me to have a connection that lives longer than the enclosing function's lifetime?

Reply:
Actually, I think for my use case, this is fine. I think I could, and should, have a connectionId initialized at the constructor of my pilot manager.